api/v1beta1/schemareplication_webhook.go

```diff
@@ -36,10 +41,25 @@ func (s *SchemaReplication) ValidateUpdate(old runtime.Object) error {
 		return apierrors.NewForbidden(s.GroupResource(), s.Name,
 			field.Forbidden(field.NewPath("spec", "rabbitmqClusterReference"), "update on rabbitmqClusterReference is forbidden"))
 	}
-	return nil
+	return s.validateSecret()
 }
 
-// no validation on delete
+// ValidateDelete no validation on delete
 func (s *SchemaReplication) ValidateDelete() error {
 	return nil
 }
+
+func (s *SchemaReplication) validateSecret() error {
+	if s.Spec.UpstreamSecret != nil && s.Spec.UpstreamSecret.Name != "" && s.Spec.SecretBackend.Vault != nil && s.Spec.SecretBackend.Vault.SecretPath != "" {
+		return apierrors.NewForbidden(s.GroupResource(), s.Name,
+			field.Forbidden(field.NewPath("spec"),
+				"do not provide both secretBackend.vault.secretPath and upstreamSecret"))
+	}
+
+	if s.Spec.UpstreamSecret == nil && s.Spec.SecretBackend.Vault == nil {
```

Comment:
This doesn't catch Vault being non-nil but with the secret unset, e.g.:

```yaml
---
apiVersion: rabbitmq.com/v1beta1
kind: SchemaReplication
metadata:
  name: upstream
spec:
  endpoints: "test:134556"
  secretBackend:
    vault: {}
  rabbitmqClusterReference:
    name: hello-world
```

This is later caught in the controller, but it's not validated upfront by this webhook.

Reply:
Good point and updated.